src/main/java/org/jabref/gui/openoffice/OOBibBaseConnect.java

```diff
@@ -0,0 +1,271 @@
+package org.jabref.gui.openoffice;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.jabref.gui.DialogService;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.openoffice.CreationException;
+import org.jabref.logic.openoffice.DocumentConnection;
+import org.jabref.logic.openoffice.NoDocumentException;
+
+import com.sun.star.comp.helper.BootstrapException;
+import com.sun.star.container.NoSuchElementException;
+import com.sun.star.container.XEnumeration;
+import com.sun.star.container.XEnumerationAccess;
+import com.sun.star.frame.XComponentLoader;
+import com.sun.star.frame.XDesktop;
+import com.sun.star.lang.WrappedTargetException;
+import com.sun.star.lang.XComponent;
+import com.sun.star.lang.XMultiComponentFactory;
+import com.sun.star.text.XTextDocument;
+import com.sun.star.uno.UnoRuntime;
+import com.sun.star.uno.XComponentContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class for manipulating the Bibliography of the currently started
+ * document in OpenOffice.
+ */
+class OOBibBaseConnect {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OOBibBaseConnect.class);
+
+    /* variables  */
+    private final DialogService dialogService;
+    private final XDesktop xDesktop;
+
+    /**
+     * Created when connected to a document.
+     *
+     * Cleared (to null) when we discover we lost the connection.
+     */
+    private DocumentConnection xDocumentConnection;
+
+    /*
+     * Constructor
+     */
+    public OOBibBaseConnect(Path loPath,
+                            DialogService dialogService)
+        throws
+        BootstrapException,
+        CreationException {
+
+        this.dialogService = dialogService;
+        this.xDesktop = simpleBootstrap(loPath);
+    }
+
+    /* *****************************
+     *
+     *  Establish connection
+     *
+     * *****************************/
+
+    private XDesktop simpleBootstrap(Path loPath)
+        throws
+        CreationException,
+        BootstrapException {
+
+        // Get the office component context:
+        XComponentContext context = org.jabref.gui.openoffice.Bootstrap.bootstrap(loPath);
+        XMultiComponentFactory sem = context.getServiceManager();
+
+        // Create the desktop, which is the root frame of the
+        // hierarchy of frames that contain viewable components:
+        Object desktop;
+        try {
+            desktop = sem.createInstanceWithContext("com.sun.star.frame.Desktop", context);
+        } catch (Exception e) {
+            throw new CreationException(e.getMessage());
+        }
+        XDesktop result = unoQI(XDesktop.class, desktop);
+
+        unoQI(XComponentLoader.class, desktop);
```

Comment:
Can this line be deleted?

Reply:
I think yes.
There were a few of these.
The [documentation](http://www.openoffice.org/udk/java/ref/com/sun/star/uno/UnoRuntime.html#queryInterface(java.lang.Class,%20java.lang.Object)) suggests that this cannot even throw.

Assuming it involves a roundtrip to OpenOffice, some of these
may had a side effect of synchronizing after one-way UNO calls.
This does not apply here, and nowadays the bridge(?) runs synchronously
unless requested otherwise.

Removed.